java/1.20.5/src/schemas/Recipe.ts

```diff
@@ -120,9 +120,19 @@ export function initRecipeSchemas(schemas: SchemaRegistry, collections: Collecti
     })
   }))
 
+  schemas.register('cooking_recipe_result', Mod(ObjectNode({
+    id: StringNode({ validator: 'resource', params: { pool: 'item' } }),
+    components: Opt(Reference('data_component_predicate'))
+  }), {
+    default: () => ({
+      id: 'minecraft:stone'
+    })
+  }))
+
   schemas.register('recipe_result', Mod(ObjectNode({
     id: StringNode({ validator: 'resource', params: { pool: 'item' } }),
-    count: Opt(Mod(NumberNode({ integer: true }), { default: () => 1 }))
+    count: Opt(Mod(NumberNode({ integer: true }), { default: () => 1 })),
+    components: Opt(Reference('data_component_predicate'))
```

Comment:
This should be `data_component_patch`, but I'm pretty sure you can replace this whole thing with `Reference('item_stack')`